src/core/Oscillator.cpp

```diff
@@ -89,6 +108,230 @@ void Oscillator::update( sampleFrame * _ab, const fpp_t _frames,
 }
 
 
+void Oscillator::generateSawWaveTable(int bands, sample_t* table, int firstBand)
+{
+	// sawtooth wave contain both even and odd harmonics
+	// hence sinewaves are added for all bands
+	// https://en.wikipedia.org/wiki/Sawtooth_wave
+	for (int i = 0; i < OscillatorConstants::WAVETABLE_LENGTH; i++)
+	{
+		// add offset to the position index to match phase of the non-wavetable saw wave; precompute "/ period"
+		const float imod = (i - OscillatorConstants::WAVETABLE_LENGTH / 2.f) / OscillatorConstants::WAVETABLE_LENGTH;
+		if (firstBand == 1) { table[i] = 0.0; }
```

Comment:
Would it be more readable if these were outside of the `generate` funcs (just one `std::fill` in the beginning)?